src/test/java/com/coherentsolutions/training/automation/java/web/urnezaite/CartTest.java

```diff
@@ -0,0 +1,32 @@
+package com.coherentsolutions.training.automation.java.web.urnezaite;
+
+import io.qameta.allure.Description;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CartTest extends BaseTest {
+
+    @Test
+    @Description("Verify the ability to add to cart")
+    public void testCart() {
+        MyAccountPage myAccountPage = logIn();
+        HomePage homePage = myAccountPage.goToHomePage();
+        List<String> itemsAddedToCart = new ArrayList<>();
+
+        while (itemsAddedToCart.size() != 3) {
+            ItemPage itemPage = homePage.selectRandomItem();
+            if (!itemsAddedToCart.contains(itemPage.getItemsReference())) {
+                String itemAddedToCart = itemPage.getItemsReference();
+                itemPage.addToCart();
+                itemsAddedToCart.add(itemAddedToCart);
+                homePage = itemPage.goToHomePage();
+            }
+        }
```

Comment:
please move this logic to a page. Tests should be as abstract and simple as possible. Even a person who does not understand anything in the automation should read the test and understand what is happening there.